test/e2e/proxy_test.go

```diff
@@ -1084,3 +1087,171 @@ func testProxyServiceLifeCycle(ipFamily *corev1.IPFamily, ingressIPs []string, d
 		}
 	}
 }
+
+// TestProxyLoadBalancerModeDSR creates a LoadBalancer Service and verifies both external and internal clients accessing
+// its LoadBalancer IPs work as expected.
+// Client IP should always be preserved regardless of whether the traffic is externally or internally originated.
+// Session affinity should take effect when it's enabled.
+func TestProxyLoadBalancerModeDSR(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+	skipIfNumNodesLessThan(t, 3)
+	skipIfAntreaIPAMTest(t)
+	skipIfProxyDisabled(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	skipIfProxyAllDisabled(t, data)
+	skipIfLoadBalancerModeIsNot(t, data, config.LoadBalancerModeDSR)
```

Comment:
maybe with the recent change (per-Service annotation override), we could relax this constraint? We just need to validate that the FeatureGate is enabled, then use an annotation on the Service we create. It could simplify the testing infra a little bit more (especially after we graduate the FG to Beta).